src/scripts/thumbnail.py

```diff
@@ -0,0 +1,179 @@
+import asyncio
+from uuid import uuid4
+from src.crud.base import CRUDBase
+from src.core.print import PrintMap
+from sqlalchemy import select, text
+from src.core.config import settings
+from src.db.models.layer import Layer
+from src.schemas.layer import LayerType
+from src.db.models.project import Project
+from src.db.session import session_manager
+from sqlalchemy.ext.asyncio import AsyncSession
+from src.db.models.system_task import SystemTask
+from datetime import datetime, timedelta, timezone
+from fastapi_pagination import Params as PaginationParams
+from src.crud.crud_user_project import user_project as crud_user_project
+from src.crud.crud_layer_project import layer_project as crud_layer_project
+
+
+async def fetch_last_run_timestamp(async_session: AsyncSession) -> datetime:
+    """Fetch the last run timestamp of the thumbnail update task."""
+
+    crud_system_task = CRUDBase(SystemTask)
+    result = await crud_system_task.get(async_session, "thumbnail_update")
+    return result.last_run if result is not None else datetime.now(timezone.utc) - timedelta(minutes=5)
+
+
+async def update_last_run_timestamp(async_session: AsyncSession, last_run: datetime):
+    """Update the last run timestamp of the thumbnail update task."""
+
+    await async_session.execute(
+        text(
+            """UPDATE customer.system_task
+            SET last_run = :last_run WHERE id = 'thumbnail_update'""",
+        ),
+        {"last_run": last_run},
+    )
+    await async_session.commit()
```

Comment:
I think here we can also use the "system_task" db model instead of raw sql. You can import it from the db.models